src/__tests__/pdfParser.test.ts

```
@@ -407,3 +407,709 @@ describe('PDFParserService', () => {
+        
+        Dialogue includes: "Hello!", "Good morning?", "Let's go..."
+        
+        The scene costs $50,000 to produce.
+      `;
+      const result = (parser as any).isExtractionSuccessful(mixedText, 1024);
+      expect(result).toBe(true);
+    });
+
+    it('rejects text with excessive numeric content (data dumps)', () => {
+      const numericText = '1234567890 '.repeat(100) + 'some text here';
+      const result = (parser as any).isExtractionSuccessful(numericText, 1024);
```

Biome (1.9.4)

[error] Template literals are preferred over string concatenation.

Unsafe fix: Use a template literal.


(lint/style/useTemplate)